models/User.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * User Model
+ *
+ * Represents a registered user in the system.
+ * Handles identity, access, and relationships to projects, comments, etc.
+ */
+
+module.exports = (sequelize, Sequelize) => {
+  const { DataTypes } = Sequelize;
+
+  const User = sequelize.define(
+    "User",
+    {
+      id: {
+        type: DataTypes.UUID,
+        defaultValue: DataTypes.UUIDV4,
+        primaryKey: true,
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      username: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        unique: true,
+        validate: {
+          len: [3, 30],
+        },
+        set(value) {
+          this.setDataValue("username", value.toLowerCase().trim());
+        },
```

Comment:
A username is probably going to be someone's display name setting this to lowercase might confuse the user and not be what they are expecting.
```suggestion
```

Reply:
True, I didn't think about how they’ll display again.It was to make sure they’re really unique (if we have case insensitive logins etc)
Could consider a username and a displayName field, if we need it for login/lookups? Otherwise I'll just remove it.
```suggestion
      username: {
        type: DataTypes.STRING,
        allowNull: false,
        unique: true,
        validate: {
          len: [3, 30],
        },
        set(value) {
          this.setDataValue("username", value.toLowerCase().trim());
        },
      },
      displayName: {
        type: DataTypes.STRING,
        allowNull: false,
      },
```

Reply:
For login we will use email so having display name is redundant when the username serves that purpose, at which point we should preserve the users desired case usage. 

Also validating with the ORM isn't great as it is slow, and will sometime make database calls for thing like uniqueness checks while doing the validation, so its better to stick to using Yup and defining our restricting on the server, not to mention the output from Yup is a lot easier to handle then what you will get from Sequilize.